sidh/sidhp751/sidhp751.go

```diff
@@ -136,9 +147,9 @@ func GenerateAliceKeypair(rand io.Reader) (publicKey *SIDHPublicKeyAlice, secret
 	}
 
 	// Bit-twiddle to ensure scalar is in 2*[0,2^371):
-	secretKey.Scalar[47] = 0
-	secretKey.Scalar[46] &= 15 // clear high bits, so scalar < 2^372
-	secretKey.Scalar[0] &= 254 // clear low bit, so scalar is even
+	secretKey.Scalar[SecretKeySize-1] = MaskAliceByte1
+	secretKey.Scalar[SecretKeySize-2] &= MaskAliceByte2 // clear high bits, so scalar < 2^372
+	secretKey.Scalar[0] &= MaskAliceByte3               // clear low bit, so scalar is even
 
```

Comment:
Question: is this masking generic or specific to p751?

Reply:
It's specific to p751 and Alice and Bob torsion subgroup order